### ISSUE TITLE
systemctl autoescaping of pathnames

### ISSUE THREAD

**Issue by bjne** [systemd/systemd#7692]
### Submission type

 - Bug report
 - Request for enhancement (RFE)

### systemd version the issue has been seen with

> 229

### Used distribution

> ununtu/xenial

### In case of bug report: Expected behaviour you didn't see

> %I == /path/to/foo-bar

### In case of bug report: Unexpected behaviour you saw

> /path/to/foo/bar

### In case of bug report: Steps to reproduce the problem

When trying to use a valid pathname with no special characters as a unit-name, systemctl
seems to auto-escape this incorrectly. I do understand that one could use systemd-escape to
produce a valid "unit-name", but I would like to argue the usability and requirement of this for
names containing no special characters. I do understand that systemd internally needs to escape
unit names for storing on the filesystem, but also feel much of this could be transparent to the
user for increased userfriendlyness :)

In my example, I am trying to start nginx with a prefix param sent to it, and created a unit file,
problem is that when I run:

# systemctl start nginx@/path/to/foo-bar

%i == -path-to-foo-bar
%I == /path/to/foo/bar

so none of these are really usable, as I can understand it, %i should have been path-to-foo\x2dbar

**Comment by yuwata** [comment]
Indeed. Your systemd version is too old. But I can reproduce this with the recent snapshot f9f5441.

**Comment by poettering** [comment]
uh, this is black magic a bit... I mean, how do we know whether the string was not escaped at all, or just badly (i.e. partly) escaped? Right now we only fix what is obviously not OK, but leave everything that validates the way it is.

I am not sure that "escape everything if there's one unescaped character" is that much better than "escape only what is not escaped"... I can easily make up reasons for both of these philosophies and I find neither trumping the other in every possible way. 

There's one major thing the current behaviour has going for it though: it has been that way since a long time, and changing the behaviour is a pretty much a compat breakage.

HenceI am not sure we should change anything here...

**Comment by bjne** [comment]
I think this issue needs more discussion, as the current impl. is really unfriendly from a users perspective.
What problems would it cause to always escape unit-name-input if it contains unescaped chars?
